File: Daily/PY/Leetcode1894-找到需要补充粉笔的学生.py
from typing import List

class Solution:
    def chalkReplacer(self, chalk: List[int], k: int) -> int:
        s = sum(chalk)
        k = k % s
        n,res = len(chalk),-1
        for i in range(n):
            if chalk[i] > k:
                res = i
                break
            else:
                k -= chalk[i]
        return res

File: Daily/sbgo/Leetcode1894-找到需要补充粉笔的学生.go
package sbgo

func chalkReplacer(chalk []int, k int) int {
	sum, n, res := 0, len(chalk), -1
	for _, val := range chalk {
		sum += val
	}
	k = k % sum
	for i := 0; i < n; i++ {
		if chalk[i] > k {
			res = i
			break
		} else {
			k -= chalk[i]
		}
	}
	return res
}
